src/loaders.ts

```diff
@@ -9,8 +9,34 @@ export type ContractResult = {
     runs: number;
 
     ok: boolean; // False if no result is found
+
+    /**
+     * getSources returns the imports -> source code mapping for the contract, if available.
+     *
+     * Caveats:
+     * - Not all loaders support this, so the property could be undefined.
+     * - This call could trigger additional fetch requests, depending on the loader.
+     **/
+    getSources?: () => Promise<ContractSources>;
 }
 
+/**
+ * ContractSources is a list of source files.
+ * If the source was flattened, it will lack a path attribute.
+ *
+ * @example
+ * ```typescript
+ * [{"content": "pragma solidity =0.7.6;\n\nimport ..."}]
+ * ```
+ *
+ * @example
+ * ```typescript
+ * [{"path": "contracts/Foo.sol", "content:" "pragma solidity =0.7.6;\n\nimport ..."}]
+ * ```
+ **/
+export type ContractSources = Array<{ path?: string, content: string }>;
```

Comment:
Alright, changed the type and added Sourcify support too, what do we think of it now?